internal/liveness/liveness.go

```diff
@@ -37,57 +44,113 @@ var (
 	})
 )
 
-func getLiveness(timeout time.Duration, csiConn *grpc.ClientConn) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+func (c *probeConn) checkProbe(w http.ResponseWriter, req *http.Request) {
+	ctx, cancel := context.WithTimeout(req.Context(), c.config.ProbeTimeout)
 	defer cancel()
 
-	util.TraceLogMsg("Sending probe request to CSI driver")
-	ready, err := rpc.Probe(ctx, csiConn)
+	util.TraceLog(ctx, "Healthz req: Sending probe request to CSI driver %s", c.config.DriverName)
+	ready, err := rpc.Probe(ctx, c.conn)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		_, err = w.Write([]byte(err.Error()))
+		if err != nil {
+			util.ErrorLog(ctx, "Healthz req: write failed: %v", err)
+		}
+		util.ErrorLog(ctx, "Healthz req: health check failed: %v", err)
```

Comment:
won't `err` value overwritten when you call w.Write()?

Reply:
good catch, will fix it.